test/tools/save-logs/extract_logs.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python
+
+'''
+Copyright 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+
+ScriptName: extract_logs.py
+Author(s): Brent Higgins, E Hohenstein
+Initial date: 09/01/16
+
+Purpose:
+This script is intended to expand the tarball created from save_logs.py
+into a target directory.
+It will create the target dir, expand tarball, and chmod 777 all the directories
+and included files to allow the user and others to easily read, add, delete as needed.
+
+'''
+import os
+import errno
+import tarfile
+import time
+import argparse
+
+import shutil
+
+
+def main():
+    ARG_PARSER = argparse.ArgumentParser(description="Command Help")
+    ARG_PARSER.add_argument("-pkg", default="None",
+                            help=".tgz file to be extracted")
+    ARG_PARSER.add_argument("-dir", default="None",
+                            help="directory .tgz file is located and default extraction location")
+    ARG_PARSER.add_argument("-output_dir", default="None",
+                            help="optional directory to extract to, defaults to tgz location")
+
+    # parse arguments to ARGS_LIST dict
+    ARGS_LIST = ARG_PARSER.parse_args()
+    if ARGS_LIST.pkg == "None":
+        print "No package given, please provide a package with the -pkg flag"
+        exit(1)
+    if ARGS_LIST.dir == "None":
+        print "No source directory given, please provide the source directory with the -dir flag"
+        exit(1)
+    if ARGS_LIST.dir[-1] != "/":
+        ARGS_LIST.dir = ARGS_LIST.dir + "/"
+    if ARGS_LIST.output_dir == "None":
+        out_dir = ARGS_LIST.dir
+    else:
+        if ARGS_LIST.output_dir[-1] != "/":
+            ARGS_LIST.output_dir = ARGS_LIST.output_dir + "/"
+        out_dir = ARGS_LIST.output_dir
+        mk_datadir(out_dir, out_dir)
+        pstr = ARGS_LIST.pkg.split("/")
+        pn = pstr[-1]
+        pkgname = pn.rstrip("\r")
+        shutil.copy(ARGS_LIST.dir + pkgname, out_dir)
+    print "Extracting to: {0}".format(out_dir)
+    extract_tgz_file_to_datadir(ARGS_LIST.pkg, out_dir)
+
+
+def extract_tgz_file_to_datadir(pkginfo, logdir):
+    '''
+    This function extracts the gathered tarball into the target directory
+    printing progress as it goes for the user.
+    :param pkginfo: string returned from gather script
+    :param logdir: full local path to bug directory
+    : return
+        True on success unpack of main tarball
+        False on any error
+    '''
+    status = False
+    pstr = pkginfo.split("/")
+    nlen = len(pstr)
+    pn = pstr[nlen - 1]
```

Comment:
If the intent is to get the last element of the list, pstr[-1] is a would better.